format-avro/src/main/java/io/cdap/plugin/format/avro/output/AvroOutputFormatProvider.java

```diff
@@ -83,22 +85,32 @@ public String getOutputFormatClassName() {
     private static final String SCHEMA_DESC = "Schema of the data to write.";
     private static final String CODEC_DESC =
       "Compression codec to use when writing data. Must be 'snappy', 'deflate', 'bzip2', 'xz', or 'none.'";
+    private static final String ALLOW_FLEXIBLE_SCHEMA =
+      "Allow records without schemas set as arguments to the pipeline.";
 
     @Macro
+    @Nullable
     @Description(SCHEMA_DESC)
     private String schema;
 
     @Macro
     @Nullable
     @Description(CODEC_DESC)
     private String compressionCodec;
+
+    @Macro
+    @Nullable
+    @Description(ALLOW_FLEXIBLE_SCHEMA)
+    private String allowFlexibleSchema;
```

Comment:
should be enough to make the schema optional, don't think we need this extra setting.

Reply:
I would argue that we should add this as a way to ensure that a missing macro somewhere doesn't cause a customer's existing pipeline to start outputting records in a different format (they expect a failure but suddenly they get records with no control over how fields appear).

Let me know what you think.

Reply:
if the schema was a macro, it had to have been filled out at runtime, otherwise the run fails right at the start. We don't allow specifying a macro and then not providing a value for it. The macro check here is only done so that it doesn't throw a NPE at validation time.